src/collection.js

```diff
@@ -525,12 +525,13 @@ export default class Collection {
       updated.last_modified = oldRecord.last_modified;
     }
     // If only local fields have changed, then keep record as synced.
+    // If status is created, keep record as created.
+    // If status is deleted, mark as updated.
     const isIdentical = oldRecord && recordsEqual(oldRecord, updated, this.localFields);
     const keepSynced = isIdentical && oldRecord._status == "synced";
-    let newStatus = (keepSynced || synced) ? "synced" : "updated";
-    if (!oldRecord) {
-      newStatus = "created";
-    }
+    const neverSynced = !oldRecord || (oldRecord && oldRecord._status == "created");
+    const newStatus = (keepSynced || synced) ? "synced"
+                                             : neverSynced ? "created" : "updated";
```

Comment:
I'm embarrassed because I think the ternary operators are actually simpler and easier to read: *«`synced` if keepSynced or synced is forced, else `created` if neverSynced else `updated`»* . 

That's a lot of code to read and process which relies boolean arithmetic (e.g. newStatus is updated when oldStatus is deleted only if never synced!).

After discussing with @n1k0, he seemed to agree also :/

Note that chaining ternary operators is not exactly prohibited, especially if the variables in the conditions are well named.